server/src/operators/message_operator.rs

```diff
@@ -611,12 +611,25 @@ pub async fn stream_response(
     let (s, r) = unbounded::<String>();
     let stream = client.chat().create_stream(parameters).await.unwrap();
 
+    let completion_first = create_message_req_payload
+        .llm_options
+        .as_ref()
+        .map(|x| x.completion_first)
+        .unwrap_or(Some(false))
+        .unwrap_or(false);
+
     Arbiter::new().spawn(async move {
         let chunk_v: Vec<String> = r.iter().collect();
         let completion = chunk_v.join("");
 
+        let message_to_be_stored = if completion_first {
+            format!("{}{}", completion, chunk_metadatas_stringified)
```

Comment:
I think it makes the most sense to store the chunks in the completion order the user wanted them to be streamed in. (Will likely work a lot better for the front ends to be modular.